File: 0071-simplify-path/0071-simplify-path.java
class Solution {
    public String simplifyPath(String path) {

        String[] arr = path.split("/");
        Stack<String> stack = new Stack<>();

        for (String i : arr) {
            if (i.equals(".") || i.equals("")) {
                continue;
            }

            
            if (i.equals("..")) {
                if (!stack.isEmpty()) {
                    stack.pop();
                }
            }

        
            else {
                stack.push(i);
            }
        }

        StringBuilder sb = new StringBuilder();

        for (String i : stack) {
            sb.append("/");
            sb.append(i);
        }

        if (sb.length() == 0) {
            return "/";
        }

        return sb.toString();
    }
}

File: 0071-simplify-path/0071-simplify-path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []

        for i in path.split("/"):
            if i == "" or i == ".":
                continue

            elif i == "..":
                if stack:
                    stack.pop()

            else:
                stack.append(i)

        return "/" + "/".join(stack)
        